File: 11-container-with-most-water/container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] nums) {
        
        int end = nums.Length-1 ;
        
        int max_area = 0 ;
        
        for(int i = 0 ; i < nums.Length ; i ++)
        {
            
            if(nums[i] <= nums[end])
            {
                max_area = Math.Max(max_area , (end - i ) * nums[i] );
            }
            else
            {
                int j = end ;
                
                while(nums[j] <= nums[i] && j > i)
                {
                    max_area = Math.Max(max_area , (j - i )*nums[j] );
                    j = j -1 ;
                }
                max_area = Math.Max(max_area , (j - i )*nums[i] );
            }
        }
        return max_area ;
            
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        max_area = 0

        while left < right:
            # Calculate current width
            width = right - left
            
            # Determine current height (limited by the shorter line)
            current_height = min(height[left], height[right])
            
            # Calculate current area
            current_area = width * current_height
            
            # Update max_area if current_area is larger
            max_area = max(max_area, current_area)
            
            # Move the pointer of the shorter line inwards
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
                
        return max_area